include/ccmath/math/nearest/rint.hpp

```diff
@@ -10,7 +10,239 @@
+	 * overloads of std::rint for all cv-unqualified floating-point types as the type of the parameter num.
+	 * @param num A floating-point value.
+	 * @return If no errors occur, the nearest integer value to num, according to the current rounding mode, is returned.
+	 */
+	constexpr double rint(double num)
+	{
+		return ccm::rint<double>(num);
+	}
+
+	/**
+	 * @brief Rounds the floating-point argument num to an integer value (in floating-point format), using the current rounding mode. The library provides
+	 * overloads of std::rint for all cv-unqualified floating-point types as the type of the parameter num.
+	 * @param num A floating-point value.
+	 * @return If no errors occur, the nearest integer value to num, according to the current rounding mode, is returned.
+	 */
+	constexpr long double rint(long double num)
+	{
+		return ccm::rint<long double>(num);
+	}
+
+	/**
+	 * @brief Rounds the floating-point argument num to an integer value (in floating-point format), using the current rounding mode. The library provides
+	 * overloads of std::rint for all cv-unqualified floating-point types as the type of the parameter num.
+	 * @param num A floating-point value.
+	 * @return If no errors occur, the nearest integer value to num, according to the current rounding mode, is returned.
+	 */
+	constexpr float rintf(float num)
+	{
+		return ccm::rint<float>(num);
+	}
+
+	/**
+	 * @brief Rounds the floating-point argument num to an integer value (in floating-point format), using the current rounding mode. The library provides
+	 * overloads of std::rint for all cv-unqualified floating-point types as the type of the parameter num.
+	 * @param num A floating-point value.
+	 * @return If no errors occur, the nearest integer value to num, according to the current rounding mode, is returned.
+	 */
+	constexpr long double rintl(long double num)
+	{
+		return ccm::rint<long double>(num);
+	}
+
+	/**
+	 * @brief Rounds the floating-point argument num to an integer value, using the current rounding mode. The library provides overloads of std::lrint and
+	 * std::llrint for all cv-unqualified floating-point types as the type of the parameter num.(
+	 * @param num A floating-point value.
+	 * @return If no errors occur, the nearest integer value to num, according to the current rounding mode, is returned.
+	 */
+	constexpr long lrint(float num)
+	{
+		return static_cast<long>(ccm::rint(num));
+	}
+
+	/**
+	 * @brief Rounds the floating-point argument num to an integer value, using the current rounding mode. The library provides overloads of std::lrint and
+	 * std::llrint for all cv-unqualified floating-point types as the type of the parameter num.(
+	 * @param num A floating-point value.
+	 * @return If no errors occur, the nearest integer value to num, according to the current rounding mode, is returned.
+	 */
+	constexpr long lrint(double num)
+	{
+		if (ccm::isnan(num))
```

Comment:
This will also suffer from the issue I mentioned earlier with `static_cast` along with every other `lrint` function that converts a float to a integer.